lib/tasks/reporting.rake

```diff
@@ -80,6 +80,47 @@ namespace :reporting do
     end
   end
 
+  desc "A CSV report of the number of publications in draft state by organisation between the dates specified [year-month-day-hour-min]"
+  task :number_of_draft_publications_by_organisation_by_date_range, %i[start_date end_date] => :environment do |_t, args|
+    if args[:start_date].present? && args[:end_date].present?
+      start_date = Time.parse(args[:start_date])
+      end_date = Time.parse(args[:end_date])
+    end
+    date_range = start_date...end_date
+
+    path = "#{Rails.root}/tmp/number_of_draft_publications_by_organisation_between_#{start_date}_to_#{end_date}.csv"
+
+    CSV_HEADERS = ["Lead publishing organisation", "Number of draft publications"].freeze
+
+    CSV.open(path, 'wb') do |csv|
+      csv << CSV_HEADERS
+
+      Edition.include(Edition::Organisations)
+
+      puts "Searching for draft publications between #{start_date} and #{end_date}"
+
+      draft_publications = Edition.latest_edition
+      .joins(:edition_organisations)
+      .where(state: "draft")
+      .where(type: "publication")
+      .where(updated_at: date_range)
+
+      organisations = Organisation.order(slug: :asc)
+
+      organisations.each do |organisation|
```

Comment:
Although it might not make a difference here, `.find_each` can often be quicker as it avoids allocating a new object for each row in the database and reuses the same one each time.